hack/build-image

```diff
@@ -203,14 +204,25 @@ def container_engine(cli):
 
 def container_build(cli, target):
     """Construct and execute a command to build the target container image."""
-    args = [container_engine(cli), "build"]
-    pkgs_from = PACKAGES_FROM[target.pkg_source]
-    if pkgs_from:
-        args.append(f"--build-arg=INSTALL_PACKAGES_FROM={pkgs_from}")
-    # docker doesn't currently support alt. architectures
+    eng = container_engine(cli)
+    args = [eng, "build"]
+
     if "docker" in args[0]:
+        # if the target arch and the host_arch are not the same, we need to use buildx
         if target.arch != host_arch():
-            raise RuntimeError("Docker does not support --arch")
+            args = [
+                eng,
+                "buildx",
+                "build",
+                f"--builder={_BUILDX_BUILDER_NAME}",
+                f"--platform=linux/{target.arch}",
+                "--load"
+            ]
+            # Docker's default builder only supports the host architecture.
+            # Therefore, we need to create a new builder to support other
+            # architectures. Errors are suppressed to prevent issues when
+            # the builder is already available - this can be improved later.
+            run(cli, [eng, "buildx", "create", f"--name={_BUILDX_BUILDER_NAME}"], check=False)
```

Comment:
I think it would be a bit clearer if we moved this line and the corresponding comment above the assignment to `args =` (currently on line 213). That way a reader immediately sees the command setting up the new builder

Reply:
Do we need to remove/clean up this builder? I won't require it for the PR but for people using the script outside the CI environment (and we exist) I wouldn't not like to leave stuff behind. Similarly, is there an issue if this command is executed >1 time on the same host?

Reply:
I added two new commits to clean up the container_build method a little bit and to address the open topics. 

When starting a new container build, the builder, which might exist due to a failure, is removed. This way, we can ensure that we always start with a fresh builder and have no leftovers from a previous build. Afterwards, a new clean builder is created. After the build process is complete, the created builder will be removed.